kong/db/dao/init.lua

```diff
@@ -1528,6 +1546,9 @@ function DAO:cache_key(key, arg2, arg3, arg4, arg5, ws_id)
       if value ~= nil then
         use_pk = false
       end
+
+      value = normalize_string(value)
```

Comment:
If we have multiple sources of `cache_key`, they will be concatenated with ":", which then may still break the limit:

https://github.com/Kong/kong/blob/13c7c6f8336075575d44ac297019f7e35bbc5145/kong/db/dao/init.lua#L1551

Should we instead do the hash at the very end? Or with more complex logic, to hash more parts of the keys until the total length is within the limit?

Reply:
This is a bit of an aggressive approach, and some functionality that was matched via cache_key prefix would be lost if we do it this way. What do you think? cc @chronolaw @dndx

Reply:
> If we have multiple sources of `cache_key`, they will be concatenated with ":",

This is the point where I am not sure if such a case exists, and if so, a test case needs to be added to cover it.

Reply:
As I know, currently we do not have this case, I think the code of @sabertobihwy is OK.